### DISCUSSION TITLE
Can you purge an instance ahead of schedule?

### DISCUSSION THREAD

**Discussion by jacobgexigo** [trimble-oss/dba-dash#118] [General]
If we added an instance for testing but want to remove it fully from the data, is there a way to purge the data prior to the purge date expiring for each context?  Would you consider adding a button or link somewhere that would allow that through the GUI?  Though I am totally fine just running a proc manually.

**Comment by DavidWiseman** [comment]
Not yet - I have thought about this though.  There is a soft delete in the GUI that you can use to remove the instance.  The data is still there - the instance is just hidden and can be restored.  

When the data is purged via the data retention settings it uses partition switching to swap the data out to a staging table and truncate it.  This is an efficient way to remove the data.  To hard-delete an instance or purge data for a specific instance prior to the retention settings would be a more resource-intensive process.  I would probably handle this via a series of batched deletes from each of the tables.  

This could be long-running in some cases so it might be better initially for this just to be a stored proc you can trigger manually.  It's also harder to delete an instance accidentally this way.  The ideal way for this to work from the GUI would be to trigger the delete by adding a message to a service broker queue.  You get an instant response from the GUI and the service broker can slowly purge the data in batches asynchronously.  Potentially you could see the delete is in progress and maybe get a notification when it's complete.  This adds some complexity over just having a stored proc that you can run manually.  

I'd definitely like to support the hard delete at some point.

**Comment by DavidWiseman** [comment]
I've created an issue from this discussion #119

**Comment by DavidWiseman** [comment]
Fix to be included in the next release. [Doc Link](https://github.com/trimble-oss/dba-dash/blob/main/Docs/FAQ.md#how-do-i-remove-an-instance)

**Comment by DavidWiseman** [comment]
Fix is in the latest release
